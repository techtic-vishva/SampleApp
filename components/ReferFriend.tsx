import React from 'react'
import { Image, StyleSheet } from 'react-native'
import { chattanoogaTapWater, dichotomousHippopotamus } from '../constants/Colors'
import OrangeButton from './OrangeButton'
import { HeaderText } from './StyledText'
import { Text, View } from './Themed'
import WhiteButton from './WhiteButton'

function ReferFriend({
	onInviteContact,
	onShare,
	showSkip,
	onSkip
}: {
	onInviteContact: () => void
	onShare: () => void
	showSkip?: boolean
	onSkip?: () => void
}) {
	return (
		<View style={styles.container}>
			<View
				style={{
					flex: 1,
					alignItems: 'center',
					justifyContent: 'center',
					width: '100%',
					backgroundColor: 'transparent'
				}}>
				<HeaderText numberOfLines={1} adjustsFontSizeToFit style={{ fontSize: 28 }}>
					Refer a Friend
				</HeaderText>
				<Text
					numberOfLines={7}
					adjustsFontSizeToFit
					style={{
						marginTop: 30,
						marginBottom: 30,
						width: '80%',
						color: chattanoogaTapWater,
						fontSize: 20,
						textAlign: 'center'
					}}>
					Help others prioritize family time by referring them to Aro.{'\n\n'}
					They will receive a discount to get started and you will earn a one month credit per referral.
				</Text>
				<WhiteButton
					outterStyle={styles.button}
					title="Invite Your Contacts"
					onPress={onInviteContact}
					icon="message-square"
					iconColorOverride={dichotomousHippopotamus}
				/>
				<WhiteButton
					outterStyle={styles.button}
					title="Share My Referral Link"
					onPress={onShare}
					icon="link"
					iconColorOverride={dichotomousHippopotamus}
				/>
			</View>

			{showSkip && (
				<OrangeButton
					outterStyle={{ width: '80%', marginTop: 'auto', marginBottom: 95 }}
					title="Continue"
					onPress={onSkip}
					icon="arrow-right"
				/>
			)}

			<Image
				style={{ position: 'absolute', top: 0, left: 0, width: '100%', height: '100%', zIndex: -1, opacity: 0.25 }}
				source={{ uri: 'https://storage.googleapis.com/aro-public/assets/photos/login-background-02.jpeg' }}
			/>
		</View>
	)
}
const styles = StyleSheet.create({
	button: {
		width: '80%',
		marginTop: 30
	},
	container: {
		flex: 1,
		alignItems: 'center'
	}
})
export default ReferFriend
