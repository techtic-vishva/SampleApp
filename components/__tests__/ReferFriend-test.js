import * as React from 'react'
import renderer from 'react-test-renderer'
import ReferFriend from '../ReferFriend'

it('render when passing the props', () => {
	const tree = renderer
		.create(
			<ReferFriend onInviteContact={() => console.log('onInviteContact')} onShare={() => console.log('onShare')} />
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
