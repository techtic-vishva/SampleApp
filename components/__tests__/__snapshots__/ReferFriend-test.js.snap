// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when passing the props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#0b0805",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#0b0805",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "objektiv",
          },
          Array [
            Object {
              "fontFamily": "objektiv-semi-bold",
              "fontSize": 20,
            },
            Object {
              "fontSize": 28,
            },
          ],
        ]
      }
    >
      Refer a Friend
    </Text>
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={7}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "objektiv",
          },
          Object {
            "color": "#999a81",
            "fontSize": 20,
            "marginBottom": 30,
            "marginTop": 30,
            "textAlign": "center",
            "width": "80%",
          },
        ]
      }
    >
      Help others prioritize family time by referring them to Aro.
      


      They will receive a discount to get started and you will earn a one month credit per referral.
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "white",
              "fontSize": 18,
            },
          ]
        }
      >
        Invite Your Contacts
      </Text>
      <View
        color="#ed832d"
        name="message-square"
        size={20}
        style={
          Array [
            Object {
              "paddingHorizontal": 10,
            },
          ]
        }
      />
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "white",
              "fontSize": 18,
            },
          ]
        }
      >
        Share My Referral Link
      </Text>
      <View
        color="#ed832d"
        name="link"
        size={20}
        style={
          Array [
            Object {
              "paddingHorizontal": 10,
            },
          ]
        }
      />
    </View>
  </View>
  <Image
    source={
      Object {
        "uri": "https://storage.googleapis.com/aro-public/assets/photos/login-background-02.jpeg",
      }
    }
    style={
      Object {
        "height": "100%",
        "left": 0,
        "opacity": 0.25,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      }
    }
  />
</View>
`;
